src/Serializer/AbstractItemNormalizer.php

```diff
@@ -147,14 +148,38 @@ protected function extractAttributes($object, $format = null, array $context = [
      */
     protected function getAllowedAttributes($classOrObject, array $context, $attributesAsString = false)
     {
+        $allowExtraAttributes = $context[self::ALLOW_EXTRA_ATTRIBUTES] ?? $this->defaultContext[self::ALLOW_EXTRA_ATTRIBUTES];
+        if (!$this->classMetadataFactory) {
+            if (!$allowExtraAttributes) {
+                throw new LogicException(sprintf('A class metadata factory must be provided in the constructor when setting "%s" to false.', self::ALLOW_EXTRA_ATTRIBUTES));
+            }
+
+            return false;
+        }
+
         $options = $this->getFactoryOptions($context);
         $propertyNames = $this->propertyNameCollectionFactory->create($context['resource_class'], $options);
 
         $allowedAttributes = [];
+        $groups = $context[self::GROUPS] ?? false;
+        $attributesInGroup = [];
+
+        if (false !== $groups) {
```

Comment:
```suggestion
        if (null !== $this->classMetadataFactory && false !== groups) {
```

You can't throw above I think it'll break things, and this dependency should be optional.

Reply:
Okay, I adopt your suggestion.